src/erc721/presets/erc721_mintable_burnable_metadata_enumerable.cairo

```diff
@@ -0,0 +1,155 @@
+            ERC721::ERC721Impl::owner_of(@unsafe_state, token_id)
+        }
+
+        fn get_approved(self: @ContractState, token_id: u256) -> ContractAddress {
+            let unsafe_state = ERC721::unsafe_new_contract_state();
+            ERC721::ERC721Impl::get_approved(@unsafe_state, token_id)
+        }
+
+        fn is_approved_for_all(
+            self: @ContractState, owner: ContractAddress, operator: ContractAddress
+        ) -> bool {
+            let unsafe_state = ERC721::unsafe_new_contract_state();
+            ERC721::ERC721Impl::is_approved_for_all(@unsafe_state, owner, operator)
+        }
+
+        fn approve(ref self: ContractState, to: ContractAddress, token_id: u256) {
+            let mut unsafe_state = ERC721::unsafe_new_contract_state();
+            ERC721::ERC721Impl::approve(ref unsafe_state, to, token_id)
+        }
+
+        fn set_approval_for_all(
+            ref self: ContractState, operator: ContractAddress, approved: bool
+        ) {
+            let mut unsafe_state = ERC721::unsafe_new_contract_state();
+            ERC721::ERC721Impl::set_approval_for_all(ref unsafe_state, operator, approved)
+        }
+
+        fn transfer_from(
+            ref self: ContractState, from: ContractAddress, to: ContractAddress, token_id: u256
+        ) {
+            let mut unsafe_state = ERC721Enumerable::unsafe_new_contract_state();
+            ERC721Enumerable::InternalImpl::transfer_from(ref unsafe_state, from, to, token_id)
+        }
+
+        fn safe_transfer_from(
+            ref self: ContractState,
+            from: ContractAddress,
+            to: ContractAddress,
+            token_id: u256,
+            data: Span<felt252>
+        ) {
+            let mut unsafe_state = ERC721Enumerable::unsafe_new_contract_state();
+            ERC721Enumerable::InternalImpl::safe_transfer_from(ref unsafe_state, from, to, token_id, data)
+        }
+    }
+
+    #[external(v0)]
+    impl ERC721MintableImpl of IERC721Mintable<ContractState> {
+        fn mint(ref self: ContractState, to: ContractAddress, token_id: u256) {
+            let mut unsafe_state = ERC721Enumerable::unsafe_new_contract_state();
+            ERC721Enumerable::InternalImpl::_mint(ref unsafe_state, to, token_id)
+        }
+    }
+
+    #[external(v0)]
+    impl ERC721BurnableImpl of IERC721Burnable<ContractState> {
+        fn burn(ref self: ContractState, token_id: u256) {
+            // [Check] Ensure that the caller is the owner of the token
+            let mut unsafe_state = ERC721::unsafe_new_contract_state();
+            let owner = ERC721::InternalImpl::_owner_of(@unsafe_state, token_id);
+            assert(get_caller_address() == owner, 'ERC721Burnable: wrong caller');
+            // [Effect] Burn the token
+            let mut unsafe_state = ERC721Enumerable::unsafe_new_contract_state();
+            ERC721Enumerable::InternalImpl::_burn(ref unsafe_state, token_id)
+        }
+    }
+
+    #[external(v0)]
```

Comment:
funnily, could we define impls in separate files and just import them?

Reply:
I deserve a try for sure, it could indeed work like Cairo 0.